### PR TITLE
Refactoring deckgl

### PR DESCRIPTION
**Pull request by betodealmeida**
@mistercrunch, publishing this early to get your thoughts (I'm planning to do it for all layers). Is this what you had in mind?

### CODE REVIEW

File: superset/assets/visualizations/deckgl/layers/index.js [hunk 1]
```diff
@@ -13,7 +12,6 @@ const layerGenerators = {
   deck_screengrid,
   deck_path,
   deck_hex,
-  deck_scatter,
```
Comment (mistercrunch) [inline superset/assets/visualizations/deckgl/layers/index.js]: removing this here may break the `deck_multi` charts that combine `deck_scatter`

**Review by mistercrunch — CHANGES_REQUESTED** [review]
I think this may break `deck_multi`, but otherwise it's the right general idea.

Looking at it now I'm thinking we may be able to keep the `layers/` folder intact, and simply get rid of the the `factory.jsx` in favor of individual visualization function for each type.

**Comment by betodealmeida** [comment]
Cool, will do. Thanks!

File: superset/assets/visualizations/main.js [hunk 2]
```diff
@@ -89,14 +86,14 @@ const vizMap = {
...
   [VIZ_TYPES.partition]: require('./partition.js'),
-  [VIZ_TYPES.deck_scatter]: deckglFactory,
-  [VIZ_TYPES.deck_screengrid]: deckglFactory,
-  [VIZ_TYPES.deck_grid]: deckglFactory,
-  [VIZ_TYPES.deck_hex]: deckglFactory,
-  [VIZ_TYPES.deck_path]: deckglFactory,
-  [VIZ_TYPES.deck_geojson]: deckglFactory,
-  [VIZ_TYPES.deck_arc]: deckglFactory,
-  [VIZ_TYPES.deck_polygon]: deckglFactory,
+  [VIZ_TYPES.deck_scatter]: require('./deckgl/layers/scatter.jsx').default,
```
Comment (betodealmeida) [inline superset/assets/visualizations/main.js]: @mistercrunch, is this the right approach? The module is now exporting two functions.

Comment (mistercrunch) [reply superset/assets/visualizations/main.js]: We should probably move to ES6 imports at some point, not a req at the moment

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**